frontend/src/lib/utils.tsx

```diff
@@ -1325,4 +1330,40 @@ export function getEventNamesForAction(actionId: string | number, allActions: Ac
         .flatMap((a) => a.steps?.filter((step) => step.event).map((step) => String(step.event)) as string[])
 }
 
+export function isPropertyGroup(
+    properties: PropertyGroupFilter | AnyPropertyFilter[] | undefined
+): properties is PropertyGroupFilter {
+    if (properties) {
+        return (properties as PropertyGroupFilter).type !== undefined
+    }
+    return false
+}
+
+export function convertPropertiesToPropertyGroup(
+    properties: PropertyGroupFilter | AnyPropertyFilter[]
+): PropertyGroupFilter {
+    if (isPropertyGroup(properties)) {
+        return properties
+    }
+    if (properties.length > 0) {
+        return { type: FilterLogicalOperator.And, values: [{ type: FilterLogicalOperator.And, values: properties }] }
+    }
+    return { type: FilterLogicalOperator.And, values: [] }
+}
+
+export function convertPropertyGroupToProperties(properties: PropertyGroupFilter | PropertyFilter[]): PropertyFilter[] {
+    if (isPropertyGroup(properties)) {
+        return flattenPropertyGroup(properties)
+    }
+    return properties
+}
+
+export function flattenPropertyGroup(properties: PropertyGroupFilter): PropertyFilter[] {
```

Comment:
quick flyby: this is too naive, since `PropertyGroup.values` can be another propertyGroup.

So, recurse, like so:
https://github.com/PostHog/posthog/blob/master/posthog/models/property.py#L191

Reply:
@EDsCODE

Reply:
I think the case where a propertyGroup might just be one layer is actually more important to consider here. We shouldn't end up with scenarios where more than two layer groups are returned to the frontend as our component couldn't handle it anyway. But, for the areas that need flattening, this util should be able to handle any depth, specifically the shallow nested groups incase some simple prop group conversion gets passed here

tl;dr we should recurse mainly to handle the shallow nesting and as a side effect will handle any nesting anyway